python/ray/experimental/noasync_serve/queues.py

```diff
@@ -0,0 +1,164 @@
+        else:
+            self.work_object_id = work_object_id
+
+
+class CentralizedQueues:
+    """A router that routes request to available workers.
+
+    Router aceepts each request from the `enqueue_request` method and enqueues
+    it. It also accepts worker request to work (called work_intention in code)
+    from workers via the `dequeue_request` method. The traffic policy is used
+    to match requests with their corresponding workers.
+
+    Behavior:
+        >>> # psuedo-code
+        >>> queue = CentralizedQueues()
+        >>> queue.enqueue_request('service-name', data)
+        # nothing happens, request is queued.
+        # returns result ObjectID, which will contains the final result
+        >>> queue.dequeue_request('backend-1')
+        # nothing happens, work intention is queued.
+        # return work ObjectID, which will contains the future request payload
+        >>> queue.link('service-name', 'backend-1')
+        # here the enqueue_requester is matched with worker, request
+        # data is put into work ObjectID, and the worker processes the request
+        # and store the result into result ObjectID
+
+    Traffic policy splits the traffic among different workers
+    probabilistically:
+
+    1. When all backends are ready to receive traffic, we will randomly
+       choose a backend based on the weights assigned by the traffic policy
+       dictionary.
+
+    2. When more than 1 but not all backends are ready, we will normalize the
+       weights of the ready backends to 1 and choose a backend via sampling.
+
+    3. When there is only 1 backend ready, we will only use that backend.
+    """
+
+    def __init__(self):
+        # service_name -> request queue
+        self.queues = defaultdict(sortedlist)
+        # service_name -> max. batch size
+        self.service_max_batch_size = {}
+        # service_name -> traffic_policy
+        self.traffic = defaultdict(dict)
+        # backend_name -> worker queue
+        self.worker_handles = defaultdict(deque)
+
+    def enqueue_request(self, service, request_data, slo=float(1e10)):
+        query = Query(request_data, slo)
+        self.queues[service].add(query)
+        self.flush()
+        return query.result_object_id
+
+    def set_max_batch(self, service, max_batch):
+        self.service_max_batch_size[service] = max_batch
+
+    def dequeue_request(self, backend, worker_handle):
+        #intention = WorkIntent()
+        #TODO: wrap all OIDs passed in lists
+        self.worker_handles[backend].append(worker_handle)
+        self.flush()
+        #return intention.work_object_id
+
+    def link(self, service, backend):
+        #logger.debug("Link %s with %s", service, backend)
+        self.traffic[service][backend] = 1.0
+        self.flush()
+
+    def set_traffic(self, service, traffic_dict):
+        #logger.debug("Setting traffic for service %s to %s", service, traffic_dict)
+        self.traffic[service] = traffic_dict
+        self.flush()
```

Comment:
This flush should happen asynchronously. Which means you need to do something like `self._router_actor_handle.flush.remote()`

Reply:
You can register's actor's handle inside it's own actor.